docker/Dockerfiles/Dockerfile-dev

```diff
@@ -1,51 +1,31 @@
-################################################################################
-# Build stage 0 `builder`:
-# Install Rally from source inside a virtualenv
-################################################################################
+FROM docker.elastic.co/wolfi/python:3.11.7-dev AS builder
```

Comment:
This is probably a good opportunity to define the rule going forward. Will we keep this version aligned with the version of the nightlies (3.11.7), or perhaps we should bump it up all the way to the newest supported version (3.12.x)?